test/parallel/test-http2-client-destroy.js

```diff
@@ -165,3 +165,32 @@ const Countdown = require('../common/countdown');
     req.on('close', common.mustCall(() => server.close()));
   }));
 }
+
+// Destroy with AbortSignal
+{
+  const server = h2.createServer();
+  const controller = new AbortController();
+
+  server.on('stream', common.mustNotCall());
+  server.listen(0, common.mustCall(() => {
+    const client = h2.connect(`http://localhost:${server.address().port}`);
+    client.on('close', common.mustCall());
+    const socket = client[kSocket];
+    socket.on('close', common.mustCall(() => {
+      assert(socket.destroyed);
+    }));
+
+    const req = client.request({}, { signal: controller.signal });
+    client.on('error', common.mustCall((err) => {
+      assert.strictEqual(err.code, 'ECONNREFUSED');
+      server.close();
```

Comment:
Add a check here to ensure that the listener has been properly removed from the `AbortSignal`